app/forms/household_relationship_form.py

```diff
@@ -17,7 +18,7 @@ def build_relationship_choices(answer_store, group_instance):
     :param group_instance: The instance of the group being iterated over
     :return:
     """
-    household_answers = answer_store.filter(block_id='household-composition')
+    household_answers = answer_store.filter(block_id='household-composition', limit=settings.EQ_MAX_NUM_REPEATS * 2)
```

Comment:
Won't it also return middle names? Maybe it would be better to query each answer_id individually?